pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -673,7 +673,6 @@ func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMoun
 			RunAsNonRoot:           &runAsNonRoot,
 			ReadOnlyRootFilesystem: &readOnlyRootFilesystem,
 		},
-		Resources: osdProps.resources,
```

Comment:
If the user is setting resources on the osds, we still should set some sort of limits on the pods. It sounds like some clusters may even have policies that require resource limits on all pods. 

What if we set the resource requests on the OSD prepare jobs to some small limit? For example 100MB memory and 1/10th CPU? 

We should do something similar for the detect version job as noted in #4069.

Reply:
Do we know how much memory `ceph-volume` and/or `lvm` expect to use? I might expect this might go up for nodes with lots of disks present.

Reply:
Most commonly we call c-v to configure one device at a time. Even when we create multiple osds with the same c-v call I'd be surprised if we're using much memory, but yeah we should measure it before adding some arbitrary limit.

Reply:
I wonder if @jan--f would have a ballpark figure?

Reply:
Hardcoding a reasonable limit for that pod looks goods to me.

Reply:
Sorry I never looked at c-v/lvm resource consumption and am not aware of anyone that has. I'd also expect this to be fairly low. One scenario where this assumption might not hold is when drives have failed. So for measuring/testing I think it'd be valuable to consider failure scenarios too.

Reply:
Speaking as an operator, who needs to tweak a nob now, and there isn't a nob to tweak, I'd prefer it be exposed via the api. When the system is broken and you determine the guessed at resource was guessed wrong, being able to override it can significantly reduce outages. Please do find a sane default. That helps everyone. But hardcoding can make an operators bad day worse.

Reply:
@kfox1111 does that latest approach solve your concern?
- by default resources are not set
- there is a new category `prepareosd` in the cluster CR resources to apply the limits just like other daemons.
- no hardcoding a value as initially proposed

Reply:
yeah, that should work.